src/app.controller.ts

```diff
@@ -66,4 +87,34 @@ export class AppController {
   getProfile(@Request() req) {
     return req.user;
   }
+
+  //set role artist
+  @ApiBearerAuth()
+  @ApiOkResponse({
+    description: 'Success to upgrade to artist',
+  })
+  @ApiConflictResponse({ description: 'User already has artist role' })
+  @HttpCode(HttpStatus.OK)
+  @HttpCode(HttpStatus.CONFLICT)
+  @Put('role/artist')
+  async upgradeToArtist(@Request() req, @Body() body: UpgradeToArtistDto) {
+    await this.userService.setRoleUser(req.user.email, Role.Artist, body);
+
+    return { message: 'success to upgrade to artist', success: true };
+  }
+
+  //set role premium
+  @ApiBearerAuth()
+  @ApiOkResponse({
+    description: 'Success to upgrade to premium',
+  })
+  @ApiConflictResponse({ description: 'User already has premium role' })
+  @HttpCode(HttpStatus.OK)
+  @HttpCode(HttpStatus.CONFLICT)
+  @Put('role/premium')
```

Comment:
Why put these role endpoints in AppController? imo, they should be in UserController